### PR TITLE
[inductor] enable Intel Compiler(icx-cl) for inductor windows

### PR DESCRIPTION
**Pull request by xuhancn**
This PR is enable Intel Compiler (`icx-cl`) for Windows inductor, likes previous PR: https://github.com/pytorch/pytorch/pull/134444 which enable clang.

Changes:
1. Fix icx-cl crash by wrong decode args, the right decode should be "utf-8".
2. Add intel compiler check, and intel compiler Windows drivers check(icx-cl).
3. Add Intel compiler openmp args config.
4. Add intel compiler openmp binary preload.

For intel compiler openmp binary path:
image

For performance, Intel compiler(`icx-cl`) is much better performance than MSVC(`cl`):
image

Append `clang-cl` performance data:
image

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/cpp_builder.py [hunk 1]
```diff
@@ -196,6 +196,33 @@ def _is_msvc_cl(cpp_compiler: str) -> bool:
     return False
 
 
+@functools.lru_cache(None)
+def _is_intel_compiler(cpp_compiler: str) -> bool:
```
Comment (jgong5) [inline torch/_inductor/cpp_builder.py:200]: Do we expect it works on Linux too or only on Windows?

Comment (xuhancn) [reply torch/_inductor/cpp_builder.py:200]: @jgong5 I think it also working on Linux, or just a small gap to fill.

**Review by jgong5 — APPROVED** [review]
We need to consider how to validate this in CI to make sure it works and not broken.

**Comment by xuhancn** [comment]
> We need to consider how to validate this in CI to make sure it works and not broken.

I will work with @chuanqi129 to do it, after msvc ci ready.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**